static/js/stream_list.js

```diff
@@ -529,14 +529,16 @@ function maybe_select_stream(e) {
     if (e.keyCode === 13) {
         // Enter key was pressed
 
-        var topStream = $('#stream_filters li.narrow-filter').first().data('name');
-        if (topStream !== undefined) {
-            // undefined if there are no results
+        var top_stream_id = $('#stream_filters li.narrow-filter').first().data('stream-id');
```

Comment:
Just as an FYI, we can eventually start using stream_sort.js for doing things like finding our first stream_id, although that code is still using names, not ids (and I haven't read forward in your PR as I write this).